src/core/ddsi/src/q_receive.c

```diff
@@ -876,14 +882,40 @@ static int handle_AckNack (struct receiver_state *rst, nn_etime_t tnow, const Ac
         }
         else
         {
-          /* no merging, send directed retransmit */
-          RSTTRACE (" RX%"PRId64"", seqbase + i);
-          enqueued = (enqueue_sample_wrlock_held (wr, seq, sample.plist, sample.serdata, prd, 0) >= 0);
-          if (enqueued)
+          /* Is this a volatile reader with a filter?
+           * If so, call the filter to see if we should re-arrange the sequence gap when needed. */
+          if (prd->filter && (prd->filter(wr, prd, sample.serdata) == 0))
```

Comment:
``prd->filter`` yields a boolean, so better use it as such. And surely there must be a more elegant way to compute ``gapstart``, ``gapend`` and the bitset without the duplication.